x/interchainstaking/keeper/receipt.go

```diff
@@ -149,7 +167,13 @@ func (k *Keeper) MintQAsset(ctx sdk.Context, sender sdk.AccAddress, senderAddres
 			},
 			uint64(ctx.BlockTime().UnixNano()+5*time.Minute.Nanoseconds()),
 		)
-	} else {
+	case mappedAddress != nil:
+		// set mapped account
+		k.SetAddressMapPair(ctx, sender, mappedAddress, zone.ChainId)
+
+		// set send to mapped account
+		err = k.BankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, mappedAddress, qAssets)
+	default:
 		err = k.BankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, sender, qAssets)
```

Comment:
@aljo242 @ajansari95 

What happens here if the account was previously mapped for this chain and but no memo was send this time around?

Conversely, what happens if no memo was sent and there is _NO_ mapping?

I think `default` here is:
```
  if zone.Is118() {
    mappedAddress, found := k.GetAddressMapPair(ctx, sender, zone.ChainId)
    if found {
      err = k.BankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, mappedAddress, qAssets)
    } else {
      // burn qAssets and return funds (ICA MsgSend back to sender)
      // perhaps work this out before the minting phase?
    }
  } else {
    err = k.BankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, sender, qAssets)
  }
```

Reply:
https://github.com/ingenuity-build/quicksilver/pull/436/commits/be99f7d8391259c0431da2c2da1a470423c22c5a
https://github.com/ingenuity-build/quicksilver/pull/436/commits/73e36c52c38a67ee0dce548c9322b40a0e15ea8a